features/poll/impl/src/main/kotlin/io/element/android/features/poll/impl/create/CreatePollPresenter.kt

```diff
@@ -0,0 +1,168 @@
+            when (event) {
+                is CreatePollEvents.Create -> scope.launch {
+                    if (canCreate) {
+                        room.createPoll(
+                            question = question,
+                            answers = answers,
+                            maxSelections = MAX_SELECTIONS,
+                            pollKind = pollKind,
+                        )
+                        // analyticsService.capture(PollCreate()) // TODO: Send PollCreate analytics.
+                        navigateUp()
+                    } else {
+                        Timber.e("Cannot create poll")
+                    }
+                }
+                is CreatePollEvents.AddAnswer -> {
+                    answers = answers + ""
+                }
+                is CreatePollEvents.RemoveAnswer -> {
+                    answers = answers.filterIndexed { index, _ -> index != event.index }
+                }
+                is CreatePollEvents.SetAnswer -> {
+                    val text = if (event.text.length > MAX_ANSWER_LENGTH) {
+                        event.text.substring(0, MAX_ANSWER_LENGTH)
+                    } else {
+                        event.text
+                    }
+                    answers = answers.toMutableList().apply {
+                        this[event.index] = text
+                    }
+                }
+                is CreatePollEvents.SetPollKind -> {
+                    pollKind = event.pollKind
+                }
+                is CreatePollEvents.SetQuestion -> {
+                    question = event.question
+                }
+                is CreatePollEvents.NavBack -> {
+                    navigateUp()
+                }
+                CreatePollEvents.ConfirmNavBack -> {
+                    val shouldConfirm = question.isNotBlank() || answers.any { it.isNotBlank() }
+                    if (shouldConfirm) {
+                        showConfirmation = true
+                    } else {
+                        navigateUp()
+                    }
+                }
+                is CreatePollEvents.HideConfirmation -> showConfirmation = false
+            }
+        }
+
+        return CreatePollState(
+            canCreate = canCreate,
+            canAddAnswer = canAddAnswer,
+            question = question,
+            answers = immutableAnswers,
+            pollKind = pollKind,
+            showConfirmation = showConfirmation,
+            eventSink = ::handleEvents,
+        )
+    }
+}
+
+private fun canCreate(
+    question: String,
+    answers: List<String>
+) = question.isNotBlank() && answers.size >= MIN_ANSWERS && answers.all { it.isNotBlank() }
+
+private fun canAddAnswer(answers: List<String>) = answers.size < MAX_ANSWERS
+
+private fun List<String>.toAnswers(): ImmutableList<Answer> {
+    return map { answer ->
+        Answer(
+            text = answer,
+            canDelete = this.size > MIN_ANSWERS,
```

Comment:
Maybe confirm with product, but I fill weird not being able to delete an option if there is only 2 options. It's still possible to add an option, then delete the undesired one, so not a big deal though.

Reply:
This was explicitly specced by Product